Set default values as "/forgot_password" in Links

babamishra9 opened the pull request:
For "create account" and "reset password" links
-Default values are set in Links model.
-Any values set in model will be reflected back to the caller.
-As per  LoginInfoEndpoint.java#L980-L991 priority of picking up
 link value for both links(/create_account, /forgot_password) is
 zone settings then global settings then default.

akarollil commented:
The commit message summary line is vague - default values for what? Also, this change is only changing the way the default password link is set, the default create account link is set as before. So what we are doing is setting default value for forgot password link like we do for create account link (consistency in default link setting), but we are also changing behavior? That's worth mentioning in the commit message too. Can we remove the line of code that resets the forgot password boolean if the forgot password link isn't provided?

babamishra9 commented:
> The commit message summary line is vague - default values for what? Also, this change is only changing the way the default password link is set, the default create account link is set as before. So what we are doing is setting default value for forgot password link like we do for create account link (consistency in default link setting), but we are also changing behavior? That's worth mentioning in the commit message too. Can we remove the line of code that resets the forgot password boolean if the forgot password link isn't provided?

@akarollil , thanks! 
Changed the commit message summary line. 

We are not changing the behaviour by the way. All use case scenarios are covered and working fine. As per  [LoginInfoEndpoint.java#L980-L991](https://github.com/GESoftware-CF/uaa/blob/dd3673f0f68439c40eb76feb0b4ab6348aacf0d0/server/src/main/java/org/cloudfoundry/identity/uaa/login/LoginInfoEndpoint.java#L980-L991) priority of picking up link value for both links(/create_account, /forgot_password) is zone settings then global settings then default. I have added same details in commit message.

[this.selfServiceResetPasswordEnabled = selfServiceResetPasswordEnabled; ](https://github.com/GESoftware-CF/uaa/pull/445/files#diff-f19237e558526f77236b6b3d7597b5d8f57489ad7853e047da1d76f53c1764dbR140) is required, if the forgot password link isn't provided.

akarollil reviewed (commented):
Looks good to me. I will let @irinaepshteyn approve.

irinaepshteyn reviewed (approved):
@babamishra9, PR looks good.  Could you refine the commit message for the second commit before merging to make it more clear.  Maybe to something like (please proof read):

Remove legacy selfServiceLinksEnabled flag from response

- Retrieving existing identity zone stored in the database with legacy
self-service flag returns response containing split self-service flags.
For example, existing configuration:
`
    "config": {
        "links": {
            "selfService": {
                "selfServiceLinksEnabled":true,
                "signup":null,
                "passwd":null
            }
        },
    }
`
yields response containing the following fragment:
`
    "config": {
        "links": {
            "selfService": {
           	"selfServiceCreateAccountEnabled": false,
           	"selfServiceResetPasswordEnabled": true,
                "signup":null,
                "passwd":"/forgot_password"
            }
        },
    }
`
- Zone provisioning request from the service broker containing legacy
self-service configuration is translated and stored in the database
with the split self-service flags. For example, service broker zone
provisioning request containing the following fragment:
`
    "config" : {
        "links" : {
            "selfService" : {
                "selfServiceLinksEnabled":true,
                "signup":"",
                "passwd":"/forgot_password"
            }
        },
    }
`
stored in the database as:
`
    "config": {
        "links": {
            "selfService": {
           	"selfServiceCreateAccountEnabled": false,
           	"selfServiceResetPasswordEnabled": true,
                "signup":"",
                "passwd":"/forgot_password"
            }
        },
    }
`

babamishra9 commented:
> @babamishra9, PR looks good. Could you refine the commit message for the second commit before merging to make it more clear. Maybe to something like (please proof read):
> 
> Remove legacy selfServiceLinksEnabled flag from response
> 
> * Retrieving existing identity zone stored in the database with legacy
>   self-service flag returns response containing split self-service flags.
>   For example, existing configuration:
>   `"config": { "links": { "selfService": { "selfServiceLinksEnabled":true, "signup":null, "passwd":null } }, }`
>   yields response containing the following fragment:
>   `"config": { "links": { "selfService": { "selfServiceCreateAccountEnabled": false, "selfServiceResetPasswordEnabled": true, "signup":null, "passwd":"/forgot_password" } }, }`
> * Zone provisioning request from the service broker containing legacy
>   self-service configuration is translated and stored in the database
>   with the split self-service flags. For example, service broker zone
>   provisioning request containing the following fragment:
>   `"config" : { "links" : { "selfService" : { "selfServiceLinksEnabled":true, "signup":"", "passwd":"/forgot_password" } }, }`
>   stored in the database as:
>   `"config": { "links": { "selfService": { "selfServiceCreateAccountEnabled": false, "selfServiceResetPasswordEnabled": true, "signup":"", "passwd":"/forgot_password" } }, }`

Done. Thanks @irinaepshteyn .
